### PR TITLE
Prepare to disable parser lookup

### PR DESCRIPTION
**Pull request by slavapestov**
This PR includes the final set of fixes before all remaining test failures are diagnostics changes. It also adds an `-enable-parser-lookup` flag to complement the existing `-disable-parser-lookup`; it becomes useful once the default is to disable parser lookup.

To deal with the test output differences, I created a second copy of each test where the output changes after disabling parser lookup. The primary copy now explicitly calls the frontend with `-disable-parser-lookup` and expects the new diagnostics; the `*_parser_lookup.swift` version calls the frontend with `-enable-parser-lookup` and has the old expectations.
    
This allows us to turn parser lookup on and off by default without disturbing tests. Once parser lookup is completely removed we can remove the `*_parser_lookup.swift` variants.

Note: Parser lookup remains *enabled* by default with this PR, except for the handful of tests where it is now explicitly disabled.

This PR builds upon https://github.com/apple/swift/pull/34135.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please test source compatibility

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: lib/AST/ASTScopeCreation.cpp [hunk 1]
```diff
@@ -831,6 +834,41 @@ void ScopeCreator::addChildrenForKnownAttributes(ValueDecl *decl,
...
+      ASTScopeAssert(endLoc.hasValue() && endLoc->isValid(),
+                     "PatternBindingDecl in local context outside of BraceStmt?");
+    }
+
+    insertionPoint =
+        ifUniqueConstructExpandAndInsert<PatternEntryDeclScope>(
+            insertionPoint, patternBinding, i,
+            isLocalBinding, endLocForBinding)
+        .getPtrOr(insertionPoint);
+
```
Comment (davidungar) [inline lib/AST/ASTScopeCreation.cpp:863]: Glad we spoke about this offline. Makes perfect sense.

**Review by davidungar — COMMENTED** [review]
Took a quick look at it, even post-merge. Looks good!

**Pull request merged**